src/python/zquantum/core/graph_test.py

```diff
@@ -117,63 +120,131 @@ def test_generate_random_regular_graph(self):
 
         self.assertEqual(len(graph.nodes), num_nodes)
 
+    def test_generate_caveman_graph(self):
+        # Given
+        number_of_cliques = 2
+        size_of_cliques = 3
+
+        # When
+        graph = generate_caveman_graph(number_of_cliques, size_of_cliques)
+
+        # Then
+        assert type(graph) == nx.Graph
```

Comment:
I would either change it to two separate tests – one for checking the type and the other for checking if the graph was created properly.

Reply:
Same for other tests.

Reply:
I got rid of this. It does not seem to bring in much value anyway.